File: Pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //Prompt
    string text = get_string("Text: ");
    //Total:
    int letter = 0;
    int word = 1;
    int sentence = 0;


    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            //Count Letter
            letter++;
        }
        if (text[i] == ' ')
        {
            //Count Word
            word++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            //Count Sentence
            sentence++;
        }
    }
    // The Coleman-Liau 
    float grade = 0.0588 * (100 * (float) letter / (float) word) - 0.296 * (100 * (float) sentence / (float) word) - 15.8;

    //Return
    if (grade > 16)
    {
        printf("Grade 16+\n");
    } 
    
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else 
    {
        printf("Grade %0.f\n", round(grade)); 
    }
}

File: pset6/readability/readability.py
# Library
from cs50 import get_string

# Prompt
text = get_string("Text: ")

# Variables
letter = 0
word = 1
sentence = 0

# Loop in text
for i in text:
    if i >= 'a' and i <= 'z' or i >= 'A' and i <= 'Z':
        letter += 1
    if i == ' ':
        word += 1
    if i == '.' or i == '!' or i == '?':
        sentence += 1
# Formula       
grade = 0.0588 * (100 * letter / word) - 0.296 * (100 * sentence / word) - 15.8

# Conditions
if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade", round(grade))